### PR TITLE
Compile and Run on Arch Linux with both python 2&3 environment 

### PR DESCRIPTION
**Pull request by godspeed1989**
This pull request fix the error i encountered with my arch Linux :)

### CODE REVIEW

File: opensfm/src/third_party/libmv/tools/tool.h [hunk 1]
```diff
@@ -34,8 +34,8 @@ namespace libmv {
 
 inline void Init(const char *usage, int *argc, char ***argv) {
   google::InitGoogleLogging((*argv)[0]);
-  google::SetUsageMessage(std::string(usage));
-  google::ParseCommandLineFlags(argc, argv, true);
+  gflags::SetUsageMessage(std::string(usage));
+  gflags::ParseCommandLineFlags(argc, argv, true);
```
Comment (godspeed1989) [inline opensfm/src/third_party/libmv/tools/tool.h]: https://github.com/search?p=2&q=SetUsageMessage&type=Code

File: opensfm/src/csfm.cc [hunk 2]
```diff
@@ -11,13 +11,22 @@
...
+  import_array();
+}
 
 BOOST_PYTHON_MODULE(csfm) {
   using namespace boost::python;
   
   google::InitGoogleLogging("csfm");
   boost::python::numeric::array::set_module_and_type("numpy", "ndarray");
-  import_array();
+  numpy_import_array_wrapper();
```
Comment (godspeed1989) [inline opensfm/src/csfm.cc]: BOOST_PYTHON_MODULE declare a function returns `void`
But `import_array()` of numpy 2&3 return different which will cause compile error.
https://github.com/numpy/numpy/blob/master/numpy/core/code_generators/generate_numpy_api.py#L136

File: opensfm/src/third_party/libmv/tools/tool.h (hunk 1 above)
Comment (paulinus) [inline opensfm/src/third_party/libmv/tools/tool.h]: Distribution packages on Ubuntu 14.4 and brew/Mac seem to come with the old google namespace. We can add
```c++
#ifndef gflags
namespace gflags=google
#endif
```
to make the code work on those platforms.

File: bin/create_tracks [hunk 3]
```diff
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
```
Comment (paulinus) [inline bin/create_tracks]: Do you know a nice way of dealing with platforms that don't have provide the python2 command yet?

Comment (godspeed1989) [inline bin/create_tracks]: not yet :(

**Comment by paulinus** [comment]
@godspeed1989 i've picked your first commit with the compatibility fixes. Thanks!

**Comment by godspeed1989** [comment]
I need to close this one in order to start a new pull request :)

**Pull request closed without merging**